### PR TITLE
fix(replays): process trace_ids / error_ids as uuids

### PR DESCRIPTION
**Pull request by JoshFerge**
- process the trace_ids / error_ids as uuids. This normalizes uuids without dashes / with dashes.
- also apply list element limits to trace_ids

### CODE REVIEW

File: snuba/datasets/replays_processor.py [hunk 1]
```diff
@@ -225,5 +235,5 @@ def process_message(
             return InsertBatch([processed], None)
         except Exception as e:
             metrics.increment("consumer_error")
-            capture_exception(e)
+            logger.exception(e)
```
Comment (cmanallen) [inline snuba/datasets/replays_processor.py:238]: `logger.exception("Replay could not be consumed.")` or something similar.  Traceback is pulled implicitly.

File: snuba/datasets/replays_processor.py [hunk 2]
```diff
@@ -81,12 +78,29 @@ def __extract_url(self, replay_event: ReplayEventDict) -> list[str]:
...
+        return [str(uuid.UUID(t)) for t in trace_ids[:LIST_ELEMENT_LIMIT]]
+
+    def __process_error_ids(self, error_ids: list[str] | None) -> list[str]:
+        if not error_ids:
+            return []
+        if len(error_ids) > LIST_ELEMENT_LIMIT:
+            metrics.increment("error_ids exceeded list limit")
+
+        return [str(uuid.UUID(e)) for e in error_ids[:LIST_ELEMENT_LIMIT]]
+
```
Comment (cmanallen) [inline snuba/datasets/replays_processor.py:96]: This section is interesting.  Clickhouse stores UUID's as 128-bit integers[1].  Normalizing the UUID should have no effect unless clickhouse preserves ingest style?  That seems unlikely though.  Are you seeing UUIDs ingested with/without dashes returning different response outputs?

1. https://github.com/ClickHouse/ClickHouse/blob/master/base/base/UUID.h#L8

**Review by cmanallen — CHANGES_REQUESTED** [review]

File: snuba/datasets/replays_processor.py (hunk 2 above)
Comment (JoshFerge) [reply snuba/datasets/replays_processor.py:96]: yes, uuids without dashes are failing for me locally. (also see we were already doing this for replay_id, and do this same processing on other uuids in other processors)

File: snuba/datasets/replays_processor.py (hunk 1 above)
Comment (JoshFerge) [reply snuba/datasets/replays_processor.py:238]: right.

File: snuba/datasets/replays_processor.py (hunk 2 above)
Comment (cmanallen) [reply snuba/datasets/replays_processor.py:96]: Makes sense to validate regardless.

**Review by cmanallen — APPROVED** [review]

File: snuba/datasets/replays_processor.py (hunk 2 above)
Comment (cmanallen) [reply snuba/datasets/replays_processor.py:96]: I see this is unrelated to the query issue.  That's right.  I remember you said that now.

Comment (JoshFerge) [reply snuba/datasets/replays_processor.py:96]: 👍🏼

**Review by evanh — APPROVED** [review]

**Pull request merged**